python/hopsworks_common/core/job_api.py

```diff
@@ -0,0 +1,272 @@
+        :param config: new job configuration
+        :type config: dict
+        :return: The updated Job object
+        :rtype: Job
+        """
+        _client = client.get_instance()
+
+        config = util.validate_job_conf(config, self._project_name)
+
+        path_params = ["project", _client._project_id, "jobs", name]
+
+        headers = {"content-type": "application/json"}
+        return job.Job.from_response_json(
+            _client._send_request(
+                "PUT", path_params, headers=headers, data=json.dumps(config)
+            )
+        )
+
+    def _schedule_job(self, name, schedule_config):
+        _client = client.get_instance()
+        path_params = ["project", _client._project_id, "jobs", name, "schedule", "v2"]
+        headers = {"content-type": "application/json"}
+        method = "PUT" if schedule_config["id"] else "POST"
+
+        return job_schedule.JobSchedule.from_response_json(
+            _client._send_request(
+                method, path_params, headers=headers, data=json.dumps(schedule_config)
+            )
+        )
+
+    def _delete_schedule_job(self, name):
+        _client = client.get_instance()
+        path_params = ["project", _client._project_id, "jobs", name, "schedule", "v2"]
+
+        return _client._send_request(
+            "DELETE",
+            path_params,
+        )
+
+    def create(
+        self,
+        name: str,
+        job_conf: Union[
+            job_configuration.JobConfiguration, ingestion_job_conf.IngestionJobConf
+        ],
+    ) -> job.Job:
+        _client = client.get_instance()
+        path_params = ["project", _client._project_id, "jobs", name]
+
+        headers = {"content-type": "application/json"}
+        return job.Job.from_response_json(
+            _client._send_request(
+                "PUT", path_params, headers=headers, data=job_conf.json()
+            )
+        )
+
+    def launch(self, name: str, args: str = None) -> None:
+        _client = client.get_instance()
+        path_params = ["project", _client._project_id, "jobs", name, "executions"]
+
+        _client._send_request("POST", path_params, data=args)
+
+    def get(self, name: str) -> job.Job:
+        _client = client.get_instance()
+        path_params = ["project", _client._project_id, "jobs", name]
+
+        return job.Job.from_response_json(_client._send_request("GET", path_params))
+
+    def last_execution(self, job: job.Job) -> execution.Execution:
+        _client = client.get_instance()
+        path_params = ["project", _client._project_id, "jobs", job.name, "executions"]
+
+        query_params = {"limit": 1, "sort_by": "submissiontime:desc"}
+
+        headers = {"content-type": "application/json"}
+        return execution.Execution.from_response_json(
+            _client._send_request(
+                "GET", path_params, headers=headers, query_params=query_params
+            ),
+            job=job,
+        )
+
+    def create_or_update_schedule_job(
```

Comment:
Well, firstly, it looks like job and schedule just should be reversed, so that instead of `create_or_update_schedule_job` and `delete_schedule_job` it will be `create_or_update_job_schedule` and `delete_job_schedule`.

Second, instead of `create_or_update` I think it can be simply `upload` or `set`, resulting in `upload_job_schedule` / `set_job_schedule`.

Reply:
attach/detach makes sense to me but I think we had a discussion during dev-ex about it being unintuitive for non native speaker. In any case I think we might need to aim for some kind of consistency with the rest of the library and decide a path forward. I think we had a similar discussion with tags for example. Let's raise it up during standup